From eca337dd4950c0a6df2a9673f84c014fb96353ef Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Wed, 8 May 2024 15:18:02 +0530
Subject: [PATCH] QSDK: mac80211: warn on once if mesh metric is zero

During performence measurement, stats is disabled through boost
performence script. So, host is not receiving the metric values
from firmware and the value becomes zero during mpath finding.
This issue addressed through static mpath. But,in RvR far range
test, connection becomes unstable and getting this warnon
continuously. So, Fixing this issue by replacing the
WARN_ON_ONCE.

Patch-dependency: none

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 net/mac80211/mesh_hwmp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/mesh_hwmp.c b/net/mac80211/mesh_hwmp.c
index f692411..1c98543 100644
--- a/net/mac80211/mesh_hwmp.c
+++ b/net/mac80211/mesh_hwmp.c
@@ -375,7 +375,7 @@ u32 airtime_link_metric_get(struct ieee80211_local *local,
 			return MAX_METRIC;
 
 		rate = ewma_mesh_tx_rate_avg_read(&sta->mesh->tx_rate_avg);
-		if (WARN_ON(!rate))
+		if (WARN_ON_ONCE(!rate))
 			return MAX_METRIC;
 
 		err = (fail_avg << ARITH_SHIFT) / 100;
-- 
2.17.1

